ledger/indexer-benchmark/src/main/scala/ledger/indexerbenchmark/IndexerBenchmark.scala

```diff
@@ -0,0 +1,227 @@
+             |  inputMapping.duration:      ${histogramToString(
+            metrics.daml.parallelIndexer.inputMapping.duration.getSnapshot
+          )}
+             |  inputMapping.duration.rate: ${metrics.daml.parallelIndexer.inputMapping.duration.getMeanRate}
+             |  ingestion.duration:         ${histogramToString(
+            metrics.daml.parallelIndexer.ingestion.duration.getSnapshot
+          )}
+             |  ingestion.duration.rate:    ${metrics.daml.parallelIndexer.ingestion.duration.getMeanRate}
+             |
+             |""".stripMargin
+        )
+
+        // Note: this allows the user to inpsect the contents of an ephemeral database
+        if (config.waitForUserInput) {
+          println(s"Index database is still running at ${config.indexerConfig.jdbcUrl}.")
+          StdIn.readLine("Press <enter> to terminate this process.")
+        }
+      }
+      Await.result(resource.asFuture, Duration(100, "hour"))
+      system.terminate()
+      ()
+    }
+  }
+
+  private[this] def createReadService(
+      updates: Iterator[(Offset, Update)],
+      config: Config,
+  ): ReadService = {
+    val initialConditions = LedgerInitialConditions(
+      IndexerBenchmark.LedgerId,
+      Configuration(
+        generation = 0,
+        timeModel = TimeModel.reasonableDefault,
+        maxDeduplicationTime = java.time.Duration.ofDays(1),
+      ),
+      Time.Timestamp.Epoch,
+    )
+
+    new ReadService {
+      override def getLedgerInitialConditions(): Source[LedgerInitialConditions, NotUsed] = {
+        Source.single(initialConditions)
+      }
+      override def stateUpdates(beginAfter: Option[Offset]): Source[(Offset, Update), NotUsed] = {
+        assert(beginAfter.isEmpty)
+        config.updateCount match {
+          case None =>
+            Source.fromIterator(() => updates)
+          case Some(updateCount) =>
+            Source
+              .cycle(() => {
+                println("(Re)starting the stream of updates")
+                updates
+              }) // TODO append-only: cycling the exact same updates is probably not a good idea
+              .take(updateCount)
+        }
+      }
+      override def currentHealth(): HealthStatus = HealthStatus.healthy
+    }
+  }
+
+  private[this] def histogramToString(data: Snapshot): String = {
+    s"[min: ${data.getMin}, median: ${data.getMedian}, max: ${data.getMax}]"
+  }
+}
+
+object IndexerBenchmark {
+  val LedgerId = "IndexerBenchmarkLedger"
+
+  def runAndExit(
+      args: Array[String],
+      updates: String => Future[Iterator[(Offset, Update)]],
+  ): Unit = {
+    val config: Config = Config.parse(args).getOrElse {
+      sys.exit(1)
+    }
+    IndexerBenchmark.runAndExit(config, updates)
+  }
+
+  def runAndExit(
+      config: Config,
+      updates: String => Future[Iterator[(Offset, Update)]],
+  ): Unit = {
+    if (config.indexerConfig.jdbcUrl.isEmpty) {
+      new IndexerBenchmark().runWithEphemeralPostgres(updates, config)
```

Comment:
Why would we need ephemeral postgres support here?

Reply:
Added a comment to `runWithEphemeralPostgres`:
```
Using an uncontrolled local database does not give good performance results, but is useful for development and functional tests.
```

Reply:
if it is just for testing, I am wondering whether we can move this into a test package.... but that is not seem to be easy, neither something we do in other case so fine by me :ok: